### DISCUSSION TITLE
Preventing a step from running on a PR originating from a fork.

### DISCUSSION THREAD

**Discussion by sandoichi** [community/community#25125] [Actions]
<p>I’m trying to figure out how to prevent a step from running on a PR workflow if the PR is coming from a fork.</p>
<p>The workflow has a few steps that should always run, like building a docker image. The last step will read repo secrets to publish the docker image to a private repo, but I don’t want that step to attempt to run if the PR came from a fork.</p>
<p>The workflow already prevents the fork-triggered-PR from accessing the secrets, and thus gets a failed step, but it would be great if the PR didn’t get a red X on it, and instead just skipped that step.</p>
<p>I tried most of the available <strong>GITHUB_</strong> environment variables, and none of them seem to provide a way to detect if the source of the PR is a fork.</p>
<p>The docs <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables" title="docs" rel="nofollow noopener">here</a> state:</p>
<table><tbody>
<tr>
<td>GITHUB_HEAD_REF</td>
<td>Only set for forked repositories. The branch of the head repository.</td>
</tr>
<tr>
<td>GITHUB_BASE_REF</td>
<td>
<p>Only set for forked repositories. The branch of the base repository.</p>
<p> </p>
</td>
</tr>
</tbody></table>
<p>But this is not true. These are getting set for workflows running on any PR, whether they came from a fork or not.</p>
<p><strong>GITHUB_REPOSITORY</strong> also seemed promising, but it ALWAYS displays the base repository, and never the fork repository.</p>
<p>Is there any way that I can have a step be skipped if the PR originated from a fork?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/sandoichi">@sandoichi</a> ,</p>
<p>There are three properties of <strong>github context</strong> you need to know:<br>
<strong>1)</strong>  <strong>github.repository</strong> – the repository where the workflow is currently running.</p>
<p><strong>2)</strong>  <strong>github.event.pull_request.base.repo.full_name</strong> – the repository where the target branch is located. It is same with <strong>github.repository</strong>.</p>
<p><strong>3)</strong>  <strong>github.event.pull_request.head.repo.full_name</strong> – the repository where the source branch is located.</p>
<p>According to the three properties:<br>
<strong>1)</strong> If the PR is merging between branches in the same repository, the value of these three properties are the same.</p>
<p><strong>2)</strong> If the PR is merging from a forked repository, the property <strong>github.event.pull_request.head.repo.full_name</strong> is different with <strong>github.event.pull_request.base.repo.full_name</strong> and <strong>github.repository</strong>.</p>
<p>So you can use these three properties to check if the PR is merging from the forked repository.</p>
<p>I have a simple example here as reference: <a href="https://github.com/ForksForTest/TestClock/blob/master/.github/workflows/Test.yml" rel="nofollow noopener">https://github.com/ForksForTest/TestClock/blob/master/.github/workflows/Test.yml</a></p>

**Comment by sandoichi** [comment]
<p><a class="mention" href="/brightran">@brightran</a>,</p>
<p><strong>github.event.pull_request.head.repo.full_name</strong> is exactly what I needed, thanks!</p>
<p>For future reference, are these particular properties documented anywhere? I didn’t see anything that listed them when starting from this page: <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions" rel="nofollow noopener">https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions</a></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sandoichi">@sandoichi</a> ,</p>
<p>Yeah, in the docs, it does not list all the possible properties from the github context, because according the different trigger events, many of the properties (and sub-properties) are different and not inherent. And also consider the number of these properties is huge. So I think it is impractical to list all possible attributes.</p>
<p>The docs only lists some main properties from the github context, and most of these main properties have their corresponding <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener"><strong>default environment variables</strong></a> automatically set by GitHub. These main properties mostly are the inherent properties in the context, no matter the workflow is triggered by what event.</p>
<p>Actually, many times we need to print the whole context information to the logs so that we can search and view the properties we need.</p>
